tests/framework/e2e/etcd_process.go

```diff
@@ -253,6 +258,43 @@ func (ep *EtcdServerProcess) Close() error {
 	return nil
 }
 
+func (ep *EtcdServerProcess) VerifySchemaVersion(lg *zap.Logger) error {
+	currentEtcdVer, err := GetVersionFromBinary(ep.cfg.ExecPath)
+	if err != nil {
+		return err
+	}
+	currentEtcdVer.Patch = 0
+	prevEtcdVer := semver.Version{Major: currentEtcdVer.Major, Minor: currentEtcdVer.Minor - 1}
+
+	dbPath := datadir.ToBackendFileName(ep.cfg.DataDirPath)
+	be := backend.NewDefaultBackend(lg, dbPath)
+	defer be.Close()
+	ver, err := schema.UnsafeDetectSchemaVersion(lg, be.BatchTx())
+	if err != nil {
+		return err
+	}
+
+	// in a mix version cluster, the storage version would be set to the cluster version,
+	// which could be lower than the server version by 1 minor version.
+	if currentEtcdVer.LessThan(ver) || ver.LessThan(prevEtcdVer) {
+		return fmt.Errorf("expect backend schema version to be between [%s, %s], but got %s", prevEtcdVer.String(), currentEtcdVer.String(), ver.String())
+	}
+	// storage schema is generally backward compatible. No need to check the buckets for higher version.
+	if ep.cfg.ExecPath == BinPath.Etcd {
+		return nil
+	}
+	lg.Info("verify no new storage schema field is present in the db file of last release process")
+	nextEtcdVer := semver.Version{Major: currentEtcdVer.Major, Minor: currentEtcdVer.Minor + 1}
+	newFields := schema.NewFieldsForVersion(nextEtcdVer)
```

Comment:
Validating consistency of schema should be a feature of schema package. Same as bbolt consistency check, schema should be able to confirm that fields presents and their values are consistent with storage version value.